### ISSUE TITLE
undefined method `helper' for ActionController::API:Class

### ISSUE THREAD

**Issue by andreibondarev** [lml/commontator#167]
Error trace when I try to start the `rails server`:

```
/Users/andrei/.rvm/gems/ruby-2.7.0@skilling-america/gems/commontator-6.2.1/lib/commontator/shared_helper.rb:62:in `block in <main>': undefined method `helper' for ActionController::API:Class (NoMethodError)
```

This commit, in my fork. fixed the issue: https://github.com/andreibondarev/commontator/commit/574b80e562cd9d0a7fbd8918da2227cd72401688

I'm using `ruby 2.7.0` and `rails 6.0.2.1`. I suspect the issue may be related to the newest version of Rails but I haven't been able to confirm it yet.

**Comment by Dantemss** [comment]
Might be related to using Rails API mode. I'll see if I can put the dummy app in API mode to test.

**Comment by andreibondarev** [comment]
@Dantemss I'm not using Rails in the API only mode.

**Comment by andreibondarev** [comment]
@Dantemss I've only looked at it briefly, but does the spec/dummy app in the test suite actually run on Rails 6 since it doesn't have it's own Gemfile?

**Comment by Dantemss** [comment]
Yes it'll run in whatever is the most recent version of each gem except bootsnap apparently because that's locked in Commontator's Gemfile.

**Comment by alaarab** [comment]
I'm actually running into this issue as well. I'm on Ruby 2.6.5 and Rails 6.0.2.1. I haven't had the time to look into what's going on, but until then, I'm on my own fork of commontator as well.

**Comment by alaarab** [comment]
@andreibondarev by any chance are you using active admin in your app? I am...

I just generated a fresh app, and I first added commontator and rspec, and ran into no issue using commontator on master branch. The second I installed ActiveAdmin, I started running into the issue. I highly doubt the issue is actually in ActiveAdmin, and do think it is linked to the code pieces you mentioned. 

 @Dantemss I notice the code pieces Andrei mentioned are referring to the same files I recently had a pull request for: https://github.com/lml/commontator/commit/fb7a7e1851aa075b47ea2306d3f4f0a3716cffc3

However, you had your own pull request which overwrote those changes, and seem to have brought this issue that Andrei and I are both experiencing in this commit: 
https://github.com/lml/commontator/commit/d5ccedb11b4fd1f4fc3c518702189b47f3780fae#diff-6fae40bbdee28bf0e67ff6b1d229d58f

Not sure if I'm totally on track here, but it might be a good idea to test adding active admin (https://activeadmin.info/documentation.html) to a dummyapp along with commontator (and maybe rspec) like I did to recreate the issue.

**Comment by alaarab** [comment]
I created PR https://github.com/lml/commontator/pull/168 here, which reverts the code mentioned in the two files 'controllers.rb' and 'shared_helper.rb' to my previous PR. This resolves the use of commontator in my application.

**Comment by andreibondarev** [comment]
@alaarab I'm using `administrate` but I did run into a `will_paginate` vs `kaminari`. Commontator uses `will_paginate` while Administrate uses `kaminari`

I did have a commit that fixes this issue as well: https://github.com/andreibondarev/commontator/commit/574b80e562cd9d0a7fbd8918da2227cd72401688

I haven't prepared the PR yet.

**Comment by Dantemss** [comment]
OK I see what's wrong...

https://github.com/rails/rails/blob/75a53297c7b7adcffef3cab0183d72969564281f/actionpack/lib/action_controller/base.rb#L270
https://github.com/rails/rails/blob/75a53297c7b7adcffef3cab0183d72969564281f/actionpack/lib/action_controller/api.rb#L149

Both of these call this hook, so the one you get depends on load order (ugh Rails).

I think the thing to do is drop Rails < 5.2 support and use the action_controller_base hook, since Rails 5.1 is no longer supported by the Rails team.

**Comment by andreibondarev** [comment]
@Dantemss Would you like me to prep a PR along with a version bump to `6.3.0`?

**Comment by alaarab** [comment]
If it works do it !

**Comment by martinstreicher** [comment]
I need these fix as well. Broke in 6.2.1.

**Comment by Dantemss** [comment]
Thanks for the PR. Fixed in 6.3.0.
